core/src/main/scala/io/finch/request/package.scala

```diff
@@ -530,6 +534,91 @@ package object request extends LowPriorityRequestReaderImplicits {
   }
 
   /**
+   * Implicit class that provides `::` and other operations on any request reader that returns a
+   * [[shapeless.HList]].
+   *
+   * See the implementation note on [[StringSeqReaderOps]] for a discussion of why this is not
+   * currently a value class.
+   */
+  final implicit class HListReaderOps[R, L <: HList](val self: PRequestReader[R, L]) {
+
+    /**
+     * Composes this request reader with the given `that` request reader.
+     */
+    def ::[S, A](that: PRequestReader[S, A])(implicit ev: S %> R): PRequestReader[S, A :: L] =
+      new PRequestReader[S, A :: L] {
+        val item = MultipleItems
+        def apply(req: S): Future[A :: L] =
+          Future.join(that(req).liftToTry, self(ev(req)).liftToTry).flatMap {
+            case (Return(a), Return(l)) => (a :: l).toFuture
+            case (Throw(a), Throw(l)) => collectExceptions(a, l).toFutureException[A :: L]
+            case (Throw(e), _) => e.toFutureException[A :: L]
+            case (_, Throw(e)) => e.toFutureException[A :: L]
+          }
+
+        def collectExceptions(a: Throwable, b: Throwable): RequestErrors = {
+          def collect(e: Throwable): Seq[Throwable] = e match {
+            case RequestErrors(errors) => errors
+            case other => Seq(other)
+          }
+
+          RequestErrors(collect(a) ++ collect(b))
+        }
+      }
+
+    /**
+     * Converts this request reader to one that returns any type with this [[shapeless.HList]] as
+     * its representation.
+     */
+    def as[A](implicit gen: Generic.Aux[A, L]): PRequestReader[R, A] = self.map(gen.from)
+
+    /**
+     * Converts this request reader to one that returns a tuple with the same types as this
+     * [[shapeless.HList]].
+     *
+     * Note that this will fail at compile time if this this [[shapeless.HList]] contains more than
+     * 22 elements.
+     */
+    def tupled(implicit tupler: Tupler[L]): PRequestReader[R, tupler.Out] = self.map(tupler(_))
```

Comment:
`tupled` is inspired by the one on Scalaz's `ApplicativeBuilder`s, but I agree that `asTuple` has a nice symmetry.